src/App.js

```diff
@@ -1,63 +1,120 @@
-// App.js
 import React, { useEffect, useState } from 'react';
 import TodoList from './TodoList';
 import AddTodoForm from './AddTodoForm';
 
 
 function App() {
-
   const [todoList, setTodoList] = useState([]);
   const [isLoading, setIsLoading] = useState(true);
 
 
+  // New async function to fetch data from the API
+  async function fetchData() {
+    const options = {
+      method: 'GET',
+      headers: {
+        Authorization: `Bearer ${process.env.REACT_APP_AIRTABLE_API_KEY}`,
+      },
+    };
 
-  useEffect(() => {
-    new Promise((resolve, reject) => {
-      setTimeout(() => {
-        resolve({
-          data: {
-            todoList: JSON.parse(localStorage.getItem("savedTodoList")) || [], // Set todoList property with data from localStorage
-          }
-        });
-      }, 2000); // Delay of 2000 milliseconds (2 seconds)
-    }).then((result) => {
-      setTodoList(result.data.todoList); // Update the todoList state with the fetched data
-      setIsLoading(false); // Turn off loading indicator
-    });
-  }, []); // Empty dependency array, so this effect runs once after initial render
-
-
-    useEffect(() => {
-      if (!isLoading) {
-        localStorage.setItem('savedTodoList', JSON.stringify(todoList)); //save the todoList as a string in the localStorage whenever it changes 
+
+    // Construct the URL for the API
+    const url = `https://api.airtable.com/v0/${process.env.REACT_APP_AIRTABLE_BASE_ID}/${process.env.REACT_APP_TABLE_NAME}`;
+
+
+    try {
+      const response = await fetch(url, options);
+
+
+      if (!response.ok) {
+        throw new Error(`Error: ${response.status}`);
```

Comment:
This is what the homework asked for and you aced it!

For future reference, this display an error like `Error: 400`. `response.status` is the HTTP status code which may not be super helpful for debugging. You could log the response in the console as well to provide more information for debugging when errors occur like:

```
console.log('Airtable error response:')
console.log(response)
throw new Error(`Error: ${response.status}`);
```

Reply:
Thank you, Elijah!